src/main/java/de/tum/in/www1/artemis/service/scheduled/DataExportScheduleService.java

```diff
@@ -0,0 +1,124 @@
+package de.tum.in.www1.artemis.service.scheduled;
+
+import java.time.*;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.DataExport;
+import de.tum.in.www1.artemis.repository.DataExportRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.DataExportCreationService;
+import de.tum.in.www1.artemis.service.ProfileService;
+
+/**
+ * Service responsible for scheduling data exports.
+ */
+@Service
+@Profile("scheduling")
+public class DataExportScheduleService {
+
+    private final ProfileService profileService;
+
+    private final TaskScheduler scheduler;
+
+    private final DataExportRepository dataExportRepository;
+
+    private final DataExportCreationService dataExportCreationService;
+
+    private final Logger log = LoggerFactory.getLogger(DataExportScheduleService.class);
+
+    public DataExportScheduleService(ProfileService profileService, @Qualifier("taskScheduler") TaskScheduler scheduler, DataExportRepository dataExportRepository,
+            DataExportCreationService dataExportCreationService) {
+        this.profileService = profileService;
+        this.scheduler = scheduler;
+        this.dataExportRepository = dataExportRepository;
+        this.dataExportCreationService = dataExportCreationService;
+
+    }
+
+    /**
+     * Schedule data exports on startup. All data export that are either in the state REQUESTED or IN_CREATION will be scheduled.
+     */
+    @PostConstruct
+    public void scheduleDataExportOnStartup() {
+        try {
+            if (profileService.isDev()) {
+                // only execute this on production server, i.e. when the prod profile is active
+                // NOTE: if you want to test this locally, please comment it out, but do not commit the changes
+                return;
+            }
+            SecurityUtils.setAuthorizationObject();
+            var dataExportsToBeScheduled = dataExportRepository.findAllThatNeedToBeScheduled();
+
+            for (var dataExport : dataExportsToBeScheduled) {
+                scheduleDataExportCreation(dataExport);
+            }
```

Comment:
Is there a reason why you schedule them on startup to run at a specific time rather than on a `Cron` scheduled task every day at 04:00 like others (e.g. cleanup) that fetches currently pending exports directly from the database at that point in time?
That would eliminate the need to keep the scheduled futures in memory for up to a day and also avoid forgetting to add requested data exports to the schedule queue via `scheduleDataExportCreation()`.

Reply:
@b-fein not really, I haven't considered using a cron scheduled task yet.
So you suggest that the scheduled job runs at a specific time and fetches all data export that are currently requested or in creation?

Reply:
Exactly, something like
```java
var dataExportsToBeScheduled = dataExportRepository.findAllThatNeedToBeScheduled();

for (var dataExport : dataExportsToBeScheduled) {
    createDataExport(dataExport);
}
```
every night.

Reply:
Example in https://github.com/ls1intum/Artemis/blob/23f27a4c0d9c1a1af9bf85c3e4e0a7feb17947b0/src/main/java/de/tum/in/www1/artemis/service/scheduled/AutomaticProgrammingExerciseCleanupService.java#L65

Reply:
@b-fein changed it to use a cron job.